sdk/src/Services/DynamoDBv2/Custom/DocumentModel/TableBuilder.cs

```diff
@@ -0,0 +1,260 @@
+            }
+            
+            if (_table.HashKeys.Count != 0)
+            {
+                throw new ArgumentOutOfRangeException("Only a single partition key is supported, and one has already been added.");
+            }
+
+            _table.HashKeys.Add(hashKeyAttribute);
+
+            _table.Keys.Add(hashKeyAttribute, new KeyDescription
+            {
+                IsHash = true,
+                Type = type
+            });
+
+            if (!_attributesAlreadyProcessed.Contains(hashKeyAttribute))
+            {
+                _table.Attributes.Add(new AttributeDefinition(hashKeyAttribute, dynamoDBEntryTypeToScalarAttributeType(type)));
+                _attributesAlreadyProcessed.Add(hashKeyAttribute);
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public ITableBuilder AddRangeKey(string rangeKeyAttribute, DynamoDBEntryType type)
+        {
+            if (string.IsNullOrEmpty(rangeKeyAttribute))
+            {
+                throw new ArgumentNullException(nameof(rangeKeyAttribute), "The name of the sort key attribute is required.");
+            }
+
+            if (_table.RangeKeys.Count != 0)
+            {
+                throw new ArgumentOutOfRangeException("Only a single sort key is supported, and one has already been added.");
+            }
+
+            _table.RangeKeys.Add(rangeKeyAttribute);
+
+            _table.Keys.Add(rangeKeyAttribute, new KeyDescription
+            {
+                IsHash = false,
+                Type = type
+            });
+
+            if (!_attributesAlreadyProcessed.Contains(rangeKeyAttribute))
+            {
+                _table.Attributes.Add(new AttributeDefinition(rangeKeyAttribute, dynamoDBEntryTypeToScalarAttributeType(type)));
+                _attributesAlreadyProcessed.Add(rangeKeyAttribute);
+            }
+
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public ITableBuilder AddLocalSecondaryIndex(string indexName, string rangeKeyAttribute, DynamoDBEntryType type)
+        {
+            if (string.IsNullOrEmpty(indexName))
+            {
+                throw new ArgumentNullException(nameof(indexName), "The name of the local secondary index is required");
+            }
+
+            if (_table.LocalSecondaryIndexes.ContainsKey(indexName))
+            {
+                throw new ArgumentException($"An local secondary index with name {indexName} has already been defined.");
+            }
+
+            if (string.IsNullOrEmpty(rangeKeyAttribute))
+            {
+                throw new ArgumentNullException(nameof(rangeKeyAttribute), "The attribute name of the range key within the local secondary index is required.");
+            }
+
+            if (_table.HashKeys.Count == 0)
+            {
+                throw new ArgumentException("A local secondary index uses the table's partition key, which was not provided. Call AddHashKey prior to AddLocalSecondaryIndex.");
```

Comment:
Be consistent and use hash key.

Reply:
Fixed in https://github.com/aws/aws-sdk-net/compare/74020f5167a3bf2e54dd53bdeba482f965e43dd2..abcc18e865203cd0aa124c2c175f0dc798d5be93